### PR TITLE
Initial implementation

### PR DESCRIPTION
**Pull request by itsmontoya**

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+	case '9':
+
+	default:
+		return false
+	}
+
+	return true
+}
+
+func isPrintableChar(r rune) (ok bool) {
```
Comment (classic-massok) [inline utils.go]: should we add `)` and `(`

File: utils.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
+package emailvalidator
+
+// Local part characters can be any of the following:
+//	- uppercase and lowercase Latin letters A to Z and a to z;
+//	- digits 0 to 9;
+//	- printable characters !#$%&'*+-/=?^_`{|}~;
```
Comment (classic-massok) [inline utils.go]: should we add `)` and `(`?

File: emailAddress_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+		{
+			emailAddress: "John.Doe@example.com",
+			err:          nil,
+		},
+		{
+			emailAddress: "engineering@hatchifyco",
+			err:          errors.Error("\"hatchifyco\" does not have a valid TLD"),
+		},
+		{
+			emailAddress: "a\"b(c)d,e:f;g<h>i[j\\k]l@example.com",
```
Comment (classic-massok) [inline emailAddress_test.go]: you have `\` in the list of printable chars.

File: utils.go (hunk 1 above)
Comment (itsmontoya) [reply utils.go]: Where are you referencing this from? I don't see parens here: https://en.wikipedia.org/wiki/Email_address#Local-part

File: utils.go (hunk 2 above)
Comment (itsmontoya) [reply utils.go]: Where are you referencing this from? I don't see parens here: https://en.wikipedia.org/wiki/Email_address#Local-part

File: emailAddress_test.go (hunk 3 above)
Comment (itsmontoya) [reply emailAddress_test.go]: `\` is an escape character

**Comment by heavenlyteam** [comment]
@itsmontoya 
Might we can use https://emailregex.com?
I used RegExp from this website onetime, and it worked.

**Pull request merged**